packages/app-desktop/gui/DefaultEditorDialog/Dialog.tsx

```diff
@@ -0,0 +1,231 @@
+	margin-right: 1em;
+	max-width: 400px;
+	opacity: 1;
+`;
+
+const SelectButton = styled(Button)`
+	padding: 10px 10px;
+    height: auto;
+    min-height: auto;
+    max-height: fit-content;
+    font-size: 1em;
+`;
+
+const ChangeEditorWarning = styled.div`
+	margin-top: 1em;
+	opacity: 0.8;
+	font-family: ${props => props.theme.fontFamily};
+	color: ${props => props.theme.color};
+	font-size: 14px;
+`;
+
+const EditorOptions: string[] = [
+	'Rich Text Editor',
+	'Markdown Editor',
+];
+
+
+const logosImageNames: Record<string, string> = {
+	'Rich Text Editor': 'default-editor_RTE_icon.png',
+	'Markdown Editor': 'default-editor_Markdown_icon.png',
+};
+
+const editorExamples: Record<string, string> = {
+	'Rich Text Editor': 'default-editor_RTE.png',
+	'Markdown Editor': 'default-editor_Markdown.png',
+};
+
+type EditorTargetInfoName = 'Rich Text Editor' | 'Markdown Editor';
+
+export default function(props: Props) {
+	const joplinCloudDescriptionRef = useRef(null);
+
+	const onButtonRowClick = useCallback(() => {
+		Setting.setValue('openDefaultEditorDialog', false);
+		props.onClose();
+	}, [props.onClose]);
+
+	const { height: descriptionHeight } = useElementSize(joplinCloudDescriptionRef);
+
+
+	const onSelectButtonClick = useCallback(async (name: EditorTargetInfoName) => {
+		const routes = {
+			'Rich Text Editor': { name: 'RTE' },
+			'Markdown Editor': { name: 'Markdown' },
+		};
+		const route = routes[name];
+
+		Setting.setValue('defaultNoteEditor', route.name);
+		Setting.setValue('openDefaultEditorDialog', false);
+		await Setting.saveAll();
+		props.onClose();
+	}, [props.onClose]);
+
+	function renderSelectArea(info: EditorInfo) {
+		return (
+			<SelectButton
+				level={ButtonLevel.Primary}
+				title={_('Select')}
+				onClick={() => onSelectButtonClick(info.name as EditorTargetInfoName)}
+				disabled={false}
+			/>
+		);
+	}
+
+	function renderEditorTarget(info: EditorInfo) {
+		const key = `editorTarget_${info.name}`;
+		const height = descriptionHeight;
+
+		const logoImageName = logosImageNames[info.name];
+		const logoImageSrc = logoImageName ? `${bridge().buildDir()}/images/${logoImageName}` : '';
+		const logo = logoImageSrc ? <EditorTargetLogo src={logoImageSrc}/> : null;
+
+		const exampleImageName = editorExamples[info.name];
+		const exampleImageSrc = exampleImageName ? `${bridge().buildDir()}/images/${exampleImageName}` : '';
+		const exampleImage = exampleImageSrc ? <EditorExample src={exampleImageSrc}/> : null;
+		const descriptionComp = <EditorTargetDescription height={height} >{info.description}</EditorTargetDescription>;
+
+		return (
+			<EditorTargetBox id={key} key={key}>
+				<EditorTargetTitle>{logo}{info.name}</EditorTargetTitle>
+				{exampleImage}
+				{descriptionComp}
+				{renderSelectArea(info)}
+			</EditorTargetBox>
+		);
+	}
+
+	function renderContent() {
+		// eslint-disable-next-line @typescript-eslint/no-explicit-any -- Old code before rule was applied
+		const boxes: any[] = [];
```

Comment:
```suggestion
		const boxes: React.ReactNode[] = [];
```
Avoid adding `eslint-disable-next-line`s in new code.